interfaces/builtin/host_docs.go

```diff
@@ -0,0 +1,76 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/mount"
+	"github.com/snapcore/snapd/osutil"
+)
+
+const hostDocsSummary = `allows access to documentation stored on the host`
+
+const hostDocsBaseDeclarationSlots = `
+  host-docs:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
```

Comment:
not super clear why this shouldn't auto-connect or why all snaps shouldn't get it by default?

Reply:
now I understand, they would normally get the base content there.  It's still a bit messy if get a proliferation of these sort of interfaces.  system-files is related but atm will not do/doesn't support any mounting.

Reply:
> not super clear why this shouldn't auto-connect or why all snaps shouldn't get it by default?

It is also an information leak since on (at least) deb-based systems this would like the entire installed package set of the host. Most applications have no use for this so making it manually connected and using a snap declaration for (eg) browsers makes sense.

Reply:
>  system-files is related but atm will not do/doesn't support any mounting.

It does not and the primary focus of the interface is to allow access to things that are already coming from the host's global mount namespace. That said, there have been times that we've allowed use of system-files for things in hostfs, which isn't too far away from supported mounts in system-files, but I think we need to be very careful here and updates to system-files would need to be thought through.

An alternative name to `host-docs` might be `system-docs`. Personally, I'm not overly concerned about adding an interface like this for system documentation-- we've gone a long time already without a proliferation of these coming up....

Reply:
Given what's in the FHS or how it is mainly populated, I can see the following names:

* system-misc-doc
* system-packages-doc  (the bits there come from distro packages, OTOH this might not be future proof if distros evolve)

notice that both my proposed names have "doc"  like the /usr/share/doc,  not "docs"